src/NServiceBus.RavenDB/Timeouts/QueryTimeouts.cs

```diff
@@ -28,6 +29,11 @@ public QueryTimeouts(IDocumentStore store)
 
         public async Task<TimeoutsChunk> GetNextChunk(DateTime startSlice)
         {
+            if(abort)
+            {
+                return new TimeoutsChunk(new TimeoutsChunk.Timeout[ 0 ], DateTime.UtcNow.AddHours(1));
```

Comment:
OK, got it. So you're saying since we already have the check in while `loop` in the worst scenario we end up issuing the query and after retrieving the first document we exit